### ISSUE TITLE
Resample training and valiation data

### ISSUE THREAD

**Issue by JasonRichard** [MIC-DKFZ/nnUNet#1347]
Dear fabian,
The original resolution of my training data is 512 × 512.
However, I want to train and validate my network in 256 × 256. 
(I know it's bad, but there is occasion when high resolution is not tolerable. like in ViTs)
Can it be done by modify `patch_size` in `nnUNetPlans.json` ?
or it just a attribute to control sliding_window train and valitate ?

Or should it be done in dataset_conversion ?

What's more, the nnUNetTrainer in nnunetv2 would not automatically create a folder when it does not exist to place the logfile.

**Comment by FabianIsensee** [comment]
Thanks for the feedback about the log file. I will look into this.
Regarding the 512 vs 256 thing there are two possible variants:
- rescale your images to 256x256 and train with 256x256
- crop 256x256 patches from 512x512 images (no rescale)
Which one do you want?
If you share your nnUNetPlans.json I can tell you what to do

**Comment by JasonRichard** [comment]
Looks like the `patch_size` attribute in `nnUNetPlans.json` does
- crop 256x256 patches from 512x512 images (no rescale)

if i want to 
- rescale images to 256x256 and train with 256x256

what should i do ?
best,

```json
{
    "dataset_name": "Dataset017_BTCV",
    "plans_name": "nnUNetPlans",
    "original_median_spacing_after_transp": [
        3.0,
        0.7578124403953552,
        0.7578124403953552
    ],
    "original_median_shape_after_transp": [
        128,
        512,
        512
    ],
    "image_reader_writer": "SimpleITKIO",
    "transpose_forward": [
        0,
        1,
        2
    ],
    "transpose_backward": [
        0,
        1,
        2
    ],
    "configurations": {
        "2d": {
            "data_identifier": "nnUNetPlans_2d",
            "preprocessor_name": "DefaultPreprocessor",
            "batch_size": 14,
            "patch_size": [
                256,
                256
            ],
            "median_image_size_in_voxels": [
                512.0,
                512.0
            ],
            "spacing": [
                0.7578124403953552,
                0.7578124403953552
            ],
            "normalization_schemes": [
                "CTNormalization"
            ],
            "use_mask_for_norm": [
                false
            ],
            "UNet_class_name": "PlainConvUNet",
            "UNet_base_num_features": 32,
            "n_conv_per_stage_encoder": ...,
            "n_conv_per_stage_decoder": ...,
            "num_pool_per_axis": ...,
            "pool_op_kernel_sizes": ...,
            "conv_kernel_sizes": ...,
            "unet_max_num_features": 512,
            "resampling_fn_data": "resample_data_or_seg_to_shape",
            "resampling_fn_seg": "resample_data_or_seg_to_shape",
            "resampling_fn_data_kwargs": {
                "is_seg": false,
                "order": 3,
                "order_z": 0,
                "force_separate_z": null
            },
            "resampling_fn_seg_kwargs": {
                "is_seg": true,
                "order": 1,
                "order_z": 0,
                "force_separate_z": null
            },
            "resampling_fn_probabilities": "resample_data_or_seg_to_shape",
            "resampling_fn_probabilities_kwargs": {
                "is_seg": false,
                "order": 1,
                "order_z": 0,
                "force_separate_z": null
            },
            "batch_dice": true
        },
        "3d_lowres": ...,
        "3d_fullres": ...,
        "3d_cascade_fullres": ...,
    "experiment_planner_used": "ExperimentPlanner",
    "label_manager": "LabelManager",
    "foreground_intensity_properties_per_channel": {
        "0": {
            "max": 3071.0,
            "mean": 80.92587280273438,
            "median": 96.0,
            "min": -1024.0,
            "percentile_00_5": -962.0,
            "percentile_99_5": 319.0,
            "std": 140.94772338867188
        }
    }
}
```

**Comment by JasonRichard** [comment]
And if i want to train with partial labels while negelecting others in the ground_truth.
For example in synapse BTCV dataset, there are 13 labels in total.
I only treasure 8 of them, are there any convenient ways to fillter them out?
Or just preprocess and mask them.

best,
